java/src/main/java/com/logicalclocks/hsfs/StorageConnector.java

```diff
@@ -155,6 +156,24 @@ public void update() throws FeatureStoreException, IOException {
     }
   }
 
+  public static class HopsFsConnector extends StorageConnector {
```

Comment:
Also this setup of so many extensions is confusing (at least to me). Not only there are many classes to check, but also end classes can have the same names, but different extensions. `HopsFsConnector` summarizes it pretty well.